klippy/extras/probe_eddy_current.py

```diff
@@ -38,7 +41,8 @@ def load_calibration(self, cal):
         self.cal_zpos = [c[1] for c in cal]
     def apply_calibration(self, samples):
         for i, (samp_time, freq, dummy_z) in enumerate(samples):
-            pos = bisect.bisect(self.cal_freqs, freq)
+            adj_freq = self.drift_comp.adjust_freq(freq, self.cal_temp)
+            pos = bisect.bisect(self.cal_freqs, adj_freq)
```

Comment:
This looks like it could cause a subtle race condition.  The `apply_calibration()` call works on a block of frequency measurements.  However, `adjust_freq()` could pull a new temperature on each individual call (as `TemperatureProbe:_temp_callback()` runs in a background thread).  That could mean that in some subtle situations a different temperature could be applied midway through a set of samples.  Probably better to consistently use a single temperature for each block of samples.

One way to do this would be to extract the temperature outside of the loop and then pass that temperature to each `adjust_freq()` call.  Another option (that might be simpler and more efficient) could be to pass all the frequencies as a list to a new `adjust_freqs()` call.

Reply:
One thing I considered was moving the call to `adjust_freq()` from `apply_calibration` to `EddyCalibration.freq_to_height()`.  This would reduce overhead from performing the adjustment on all samples, with the downside that the adjustment would't be applied to the z values in the dump command.

That said, I'm fine with capturing the probe temperature outside of the loop and applying it to all samples.

Reply:
I think it would be helpful if the API server continued to export a Z value that is temperature adjusted.